### PR TITLE
Including Plotly functions

### PR DESCRIPTION
**Pull request by liuyingh**

### CODE REVIEW

**Comment by shajoezhu** [comment]
@liuyingh  great job! 

can you improve this pull request by
1. follow lintr instructions to improve the coding style, ignore snake_case.
2. write test cases for these code

File: DESCRIPTION [hunk 1]
```diff
@@ -22,7 +22,9 @@ Depends:
     R (>= 3.1.0)
 Imports:
     Rcpp (>= 0.11.2),
-    scales (>= 0.4.0)
+    scales (>= 0.4.0),
+    plotly (>= 4.7.1),
+    magrittr
```
Comment (shajoezhu) [inline DESCRIPTION]: version

File: NAMESPACE [hunk 2]
```diff
@@ -15,3 +17,8 @@ export(histWSAF)
 export(extractPLAF)
 export(plotWSAFvsPLAF)
 export(plotObsExpWSAF)
+export(plotAltVsRefPlotly)
+export(plotHistWSAFPlotly)
+export(plotWSAFVsPLAFplotly)
+export(plotObsExpWSAFPlotly)
+
```
Comment (shajoezhu) [inline NAMESPACE]: remove last line

File: R/deploidPlotly.R [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+           yaxis = list(title = "Expected WSAF (h%*%p)", range = c(0,1),
+                        titlefont = list(size = 18, color = "black"),
+                        tickfont = list(size = 16, color = "black")),
+           shapes = list(list(type = "line", fillcolor = "black",
+                              line = list(color = "black",
+                                          width = 1.2, dash = "dot"),
+                              opacity = 0.8, x0 = 0, x1 = 1, y0 = 0, y1 = 1)))
+}
+
+
```
Comment (shajoezhu) [inline R/deploidPlotly.R]: remove empty lines

**Comment by shajoezhu** [comment]
rename the file R/deploidPlotly.R to R/dEploidPlotly.R

**Comment by shajoezhu** [comment]
add test functions
tests/test-dEploidPlotly.R
```R
context("...")
...
```

File: DESCRIPTION (hunk 1 above)
Comment (shajoezhu) [inline DESCRIPTION]: @liuyingh can you update the version here please

**Comment by shajoezhu** [comment]
@liuyingh can you rename the file deploidPlotly.R to dEploidPlotly.R

File: tests/testthat/test-DEploid_tools.R [hunk 4]
```diff
@@ -44,15 +44,25 @@ test_that("WSAF Related", {
     obsWSAF <- computeObsWSAF(PG0390CoverageVcf$altCount,
```
Comment (shajoezhu) [inline tests/testthat/test-DEploid_tools.R]: do something like this

```R
test_that("WSAF Related", {
    obsWSAF <- computeObsWSAF(PG0390CoverageVcf$altCount,
        PG0390CoverageVcf$refCount)
    expect_that(histWSAF(obsWSAF), is_a("histogram"))
    png("histWSAF.png")
    histWSAF(obsWSAF)
    dev.off()
    expect_that(inherits(plotHistWSAFPlotly(obsWSAF), "plotly"), is_true())
    p <- plotHistWSAFPlotly(obsWSAF)
    htmlwidgets::saveWidget(p, file = "histWSAF.html")
    expect_null(plotWSAFvsPLAF(plaf, obsWSAF))
    png("WSAFvsPLAF.png")
    plotWSAFvsPLAF(plaf, obsWSAF)
    dev.off()
    expect_null(plotWSAFvsPLAF(plaf, obsWSAF, expWSAF))
    expect_that(inherits(plotWSAFVsPLAFPlotly(plaf, obsWSAF,
                                              PG0390CoverageVcf$refCount,
                                              PG0390CoverageVcf$altCount),
                         "plotly"), is_true())
    p <- plotWSAFVsPLAFPlotly(plaf, obsWSAF,
                                              PG0390CoverageVcf$refCount,
                                              PG0390CoverageVcf$altCount)
    htmlwidgets::saveWidget(p, file = "WSAFvsPLAF.html")
    expect_null(plotObsExpWSAF(obsWSAF, expWSAF))
    expect_that(inherits(plotObsExpWSAFPlotly(obsWSAF, expWSAF),
                         "plotly"), is_true())
})
```

so you will be able to compare the images when you test your DEploid-r package locally, see what the images look like. You will need to add htmlwidgets into dependencies in the DESCRIPTION file
image
image

File: R/deploidPlotly.R [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+#' altFile = system.file("extdata", "PG0390-C.test.alt", package = "DEploid")
+#' PG0390CoverageT = extractCoverageFromTxt(refFile, altFile)
+#' plotAltVsRefPlotly(PG0390CoverageT$refCount, PG0390CoverageT$altCount)
+#'
+#' # Example 2
+#' vcfFile = system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
+#' PG0390CoverageV = extractCoverageFromVcf(vcfFile)
+#' plotAltVsRefPlotly(PG0390CoverageV$refCount, PG0390CoverageV$altCount)
+#'
+plotAltVsRefPlotly <- function(ref, alt){
```
Comment (shajoezhu) [inline R/deploidPlotly.R]: see https://github.com/mcveanlab/DEploid-r/blob/master/R/DEploidR.R#L200

title is an input as well

File: R/deploidPlotly.R [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+#' myhist = plotHistWSAFPlotly(obsWSAF)
+#'
+#' # Example 2
+#' vcfFile = system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
+#' PG0390CoverageV = extractCoverageFromVcf(vcfFile)
+#' obsWSAF = computeObsWSAF(PG0390CoverageV$altCount, PG0390CoverageV$refCount)
+#' plotHistWSAFPlotly(obsWSAF)
+#' myhist = plotHistWSAFPlotly(obsWSAF)
+#'
+plotHistWSAFPlotly <- function(obsWSAF){
```
Comment (shajoezhu) [inline R/deploidPlotly.R]: consider the input, add boolean variable, to make the histogram strictly 0< wsaf <1

title is also an input

https://github.com/mcveanlab/DEploid-r/blob/master/R/DEploidR.R#L266

File: R/deploidPlotly.R [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+#' vcfFile = system.file("extdata", "PG0390-C.test.vcf.gz", package = "DEploid")
+#' PG0390CoverageV = extractCoverageFromVcf(vcfFile)
+#' obsWSAF = computeObsWSAF(PG0390CoverageV$altCount, PG0390CoverageV$refCount)
+#' plafFile = system.file("extdata", "labStrains.test.PLAF.txt",
+#'   package = "DEploid")
+#' plaf = extractPLAF(plafFile)
+#' plotWSAFVsPLAFPlotly(plaf, obsWSAF, PG0390CoverageV$refCount,
+#'                PG0390CoverageV$altCount)
+#'
+plotWSAFVsPLAFPlotly <- function(plaf, obsWSAF, ref, alt){
```
Comment (shajoezhu) [inline R/deploidPlotly.R]: make title as input

File: R/deploidPlotly.R [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+#'                                        "-plaf", plafFile, "-noPanel"))
+#'
+#' prop = PG0390CoverageV.deconv$Proportions[dim(PG0390CoverageV.deconv
+#'                                               $Proportions)[1],]
+#'
+#' expWSAF = t(PG0390CoverageV.deconv$Haps) %*% prop
+#' plotObsExpWSAFPlotly(obsWSAF, expWSAF)
+#' }
+#'
+plotObsExpWSAFPlotly <- function(obsWSAF, expWSAF){
```
Comment (shajoezhu) [inline R/deploidPlotly.R]: make title as input

File: tests/testthat/test-DEploid_tools.R (hunk 4 above)
Comment (liuyingh) [inline tests/testthat/test-DEploid_tools.R]: Histogram OK
histwsaf
newplot

Comment (liuyingh) [inline tests/testthat/test-DEploid_tools.R]: Alt Vs Ref OK
altvsref
newplot 1

Comment (liuyingh) [inline tests/testthat/test-DEploid_tools.R]: Alt Vs Ref OK
altvsref
newplot 3

Comment (liuyingh) [inline tests/testthat/test-DEploid_tools.R]: Obs vs Exp OK
obsexpwsaf
newplot 4

**Pull request merged**